server/controllers/user.js

```diff
@@ -10,14 +10,31 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
   let users;
   if (searchString) {
     users = await User.find({
-      username: { $regex: searchString, $options: "i" }
+      username: { $regex: searchString, $options: 'i' },
     });
   }
 
   if (!users) {
     res.status(404);
-    throw new Error("No users found in search");
+    throw new Error('No users found in search');
   }
 
   res.status(200).json({ users: users });
 });
+
+// @route POST /users/info
+// @desc Update user personal info
+// @access Public
+exports.updatePersonalInformation = asyncHandler(async (req, res, next) => {
+  const { email, headline, bio, location } = req.body;
+  try {
+    // const user = User.findOneAndUpdate({ email }, { headline: headline, bio: bio, location: location }, { new: true });
+    const user = await User.updateOne({ email }, { headline: headline, bio: bio, location: location });
```

Comment:
Maybe consider returning the updated User document so that we can update the FE right away?

Also, please see link for difference between updateOne and findOneAndUpdate: https://stackoverflow.com/questions/31808786/mongoose-difference-of-findoneandupdate-and-update

Reply:
I changed the update to use findOneAndUpdate (my problem was that I didn't have the await command in there.
I'm returning the user object now, but have a question.  The current auth context does not have an pure update user function. 'updateLoginContext' is expecting a token etc, so should I create that update functionality in 'useAuthContext' or should someone else, then I'll add the update.

Reply:
I noticed that the User interface expects a token, at the moment we are not using the token on the FE (You can choose to leave this as is or mark the token field as an optional one and return the User document to update the local state)